### PR TITLE
demo code of protected session

### PR DESCRIPTION
**Pull request by lingcui8**
Added demo code about how to create/destroy protected session and get
FW version

### CODE REVIEW

**Comment by shiqiangyu** [comment]
Test.

File: vendor/intel/protectedsessiondemo.c [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+    VAConfigID config_id = 0;
+    int major_ver = 0;
+    int minor_ver = 0;
+    VADisplay va_dpy = NULL;
+    VAStatus va_status = VA_STATUS_ERROR_OPERATION_FAILED;
+    VAProtectedSessionID protected_session;
+    uint16_t version = 0xFF;
+    VABufferID buffer = 0;
+    VAProtectedSessionExecuteBuffer exec_buff = {0};
+    uint8_t data_get_fw[16] = {0x02, 0x00, 0x04, 0x00, 0x28, 0x00, 0x00, 0x00,
```
Comment (lindongw) [inline vendor/intel/protectedsessiondemo.c]: change uint32_t should be better.

File: vendor/intel/protectedsessiondemo.c [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+#include <va/va_prot.h>
+#include "va_display.h"
+
+#define CHECK_VASTATUS(va_status,func)                                  \
+if (va_status != VA_STATUS_SUCCESS) {                                   \
+    fprintf(stderr,"%s: failed with error 0x%x, exit\n", func, va_status); \
+    exit(1);                                                            \
+}
+
+#define FW_VERSION_OFFSET 36
```
Comment (lindongw) [inline vendor/intel/protectedsessiondemo.c]: csme fw version, should be ApiVersion in cmd header, not pavp version,

File: vendor/intel/protectedsessiondemo.c [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    int minor_ver = 0;
+    VADisplay va_dpy = NULL;
+    VAStatus va_status = VA_STATUS_ERROR_OPERATION_FAILED;
+    VAProtectedSessionID protected_session;
+    uint16_t version = 0xFF;
+    VABufferID buffer = 0;
+    VAProtectedSessionExecuteBuffer exec_buff = {0};
+    uint8_t data_get_fw[16] = {0x02, 0x00, 0x04, 0x00, 0x28, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00};
+    uint8_t output_get_fw[92];
```
Comment (lindongw) [inline vendor/intel/protectedsessiondemo.c]: try avoid hardcode

File: vendor/intel/protectedsessiondemo.c [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        sizeof(exec_buff),
+        1,
+        &exec_buff,
+        &buffer);
+    CHECK_VASTATUS(va_status, "vaCreateBuffer");
+    va_status = vaProtectedSessionExecute(va_dpy, protected_session, buffer);
+    CHECK_VASTATUS(va_status, "vaProtectedSessionExecute");
+    vaDestroyBuffer(va_dpy, buffer);
+
+    printf("\nFW version %x\n", *(uint32_t *)(output_get_fw + FW_VERSION_OFFSET));
```
Comment (lindongw) [inline vendor/intel/protectedsessiondemo.c]: don't need cast if use uint32_t

**Review by lindongw — CHANGES_REQUESTED** [review]

File: vendor/intel/protectedsessiondemo.c (hunk 1 above)
Comment (lingcui8) [reply vendor/intel/protectedsessiondemo.c]: sure, thanks

File: vendor/intel/protectedsessiondemo.c (hunk 2 above)
Comment (lingcui8) [reply vendor/intel/protectedsessiondemo.c]: In ouput, apiversion is not changed and same as input, so will keep use pavp version as offline sync

File: vendor/intel/protectedsessiondemo.c (hunk 3 above)
Comment (lingcui8) [reply vendor/intel/protectedsessiondemo.c]: sure, thanks

File: vendor/intel/protectedsessiondemo.c (hunk 4 above)
Comment (lingcui8) [reply vendor/intel/protectedsessiondemo.c]: sure, thanks

**Pull request closed without merging**